src/main/java/com/exasol/adapter/dialects/sqlserver/SqlServerSqlGenerationVisitor.java

```diff
@@ -70,61 +59,45 @@ protected String representAsteriskInSelectList(final SqlSelectList selectList) t
     }
 
     private List<String> buildSelectStar(final SqlSelectList selectList) throws AdapterException {
-
-        if (SqlGenerationHelper.selectListRequiresCasts(selectList, this.nodeRequiresCast)) {
-            return buildSelectStarWithNodeCast(selectList);
-        } else {
-            return new ArrayList<>(Collections.singletonList("*"));
-        }
+        final Optional<List<String>> selectStartWithCastedColumns = buildSelectStarWithNodeCast(selectList);
+        return selectStartWithCastedColumns.orElseGet(() -> new ArrayList<>(Collections.singletonList("*")));
     }
 
-    private List<String> buildSelectStarWithNodeCast(final SqlSelectList selectList) throws AdapterException {
+    private Optional<List<String>> buildSelectStarWithNodeCast(final SqlSelectList selectList) throws AdapterException {
+        boolean requiresCast = false;
         final SqlStatementSelect select = (SqlStatementSelect) selectList.getParent();
         int columnId = 0;
         final List<TableMetadata> tableMetadata = new ArrayList<>();
         SqlGenerationHelper.addMetadata(select.getFromClause(), tableMetadata);
         final List<String> selectListElements = new ArrayList<>(tableMetadata.size());
         for (final TableMetadata tableMeta : tableMetadata) {
             for (final ColumnMetadata columnMeta : tableMeta.getColumns()) {
+                if (requiresCast(new SqlColumn(columnId, columnMeta))) {
+                    requiresCast = true;
+                }
                 final SqlColumn sqlColumn = new SqlColumn(columnId, columnMeta);
-                selectListElements.add(buildColumnProjectionString(sqlColumn, super.visit(sqlColumn)));
+                selectListElements.add(buildColumnProjectionString(getJdbcDataType(sqlColumn), super.visit(sqlColumn)));
                 ++columnId;
             }
         }
-        return selectListElements;
+        return requiresCast ? Optional.of(selectListElements) : Optional.empty();
     }
 
-    private String buildColumnProjectionString(final SqlColumn column, final String projectionString)
-            throws AdapterException {
-        return buildColumnProjectionString(getTypeNameFromColumn(column), projectionString);
+    private boolean requiresCast(final SqlColumn column) {
+        final int typeName = getJdbcDataType(column);
+        return REQUIRE_CAST.contains(typeName);
     }
 
-    private final java.util.function.Predicate<SqlNode> nodeRequiresCast = node -> {
+    protected int getJdbcDataType(final SqlColumn column) {
```

Comment:
Not sure if we add javadoc to protected methods.

Reply:
if we do also check the other methods in this class

Reply:
Made the method private!